### ISSUE TITLE
DataFetcher Unit Tests Unable To Pass Request Headers

### ISSUE THREAD

**Issue by dpatra1** [Netflix/dgs-framework#1417]
I am trying to write unit test cases for my Datafetcher where I need a request header. But unfortunately, the header from the test is not being passed, its always null inside DatFetcher when I am executing the tests.

Does anyone has faced this similar issue? Do we have a solution for this?

**DataFetcher**
  
  `@DgsQuery public CustomResult pullData( @InputArgument CustomRequest req, @RequestHeader("CLIENT_ID") String clientId) { return new CustomResult( service.pullData(req, clientId)); }`

**UnitTest For DateFetcher**
` public static final String QUERY =
"{\n"
+ " pulldata(hsExamPreSignedUrlReq: {fileName : "ABC.pdf"}) {\n"
+ " customResultMetadata {\n"
+ " name\n"
+ " }\n"
+ " }\n"
+ "}";
`
`
@test
void pullDataTest() {
MultiValueMap<String, String> map = new LinkedMultiValueMap<>();
map.add(CLIENT_ID, UUID.randomUUID().toString());
HttpHeaders httpHeaders = new HttpHeaders(map);

CustomResult response = new CustomResult(clientId, name);

String value =
    dgsQueryExecutor.executeAndExtractJsonPath(
        QUERY,
        "data.pulldata.customResultMetadata.name",
        httpHeaders);
        
assertThat(value).isEqualTo("test");
}`

When the test is getting executed, @RequestHeader("CLIENT_ID") String clientId value is going as null.

Thanks In Advance
Deba

**Comment by srinivasankavitha** [comment]
We'll look into fixing the original API. In the meantime, could you create a mock request and set the headers there, passing it in? This would be a workaround for your set up:
```

val request = MockHttpServletRequest().addHeader("foo", "bar")

executor.execute(
            query,
            variables,
            emptyMap(),
            HttpHeaders.EMPTY,
            null,
            ServletWebRequest(request)
)
```

**Comment by kilink** [comment]
This issue should be resolved with #1421.

**Comment by dpatra1** [comment]
What is the target version when this bug fix will be released? I would target updating to that version.

**Comment by GoodNello** [comment]
Hello @srinivasankavitha, is this going to be fixed also for the `v5.5` branch? Does not seem to be working on `5.5.3`

**Comment by srinivasankavitha** [comment]
Yes, most bug fixes (unless a security patch or major feature) will be available on 6.x and greater releases going forward. We expect to deprecate 5.5.x towards the end of year or early next year for maintainability.
